### PR TITLE
Use orbVersionRef query when fetching source

### PR DESCRIPTION
**Pull request by zzak**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -730,28 +730,42 @@ func OrbPromote(ctx context.Context, logger *logger.Logger, namespace string, or
 	return &response.PromoteOrb.OrbPromoteResponse.Orb, err
 }
 
+func orbVersionRef(orb string) string {
+	split := strings.Split(orb, "@")
+	if len(split) > 1 {
+		return strings.Join(split, "@")
+	}
+
+	// If no version was supplied, prepend @volatile to the reference
```
Comment (DanToml) [inline api/api.go]: The comment says prepend, but this is appending?

File: api/api.go [hunk 2]
```diff
@@ -730,28 +730,42 @@ func OrbPromote(ctx context.Context, logger *logger.Logger, namespace string, or
 	return &response.PromoteOrb.OrbPromoteResponse.Orb, err
 }
 
+func orbVersionRef(orb string) string {
```
Comment (DanToml) [inline api/api.go]: This is a fairly simple function, but it might be worth adding a couple of test cases.

File: api/api.go [hunk 3]
```diff
@@ -730,28 +730,42 @@ func OrbPromote(ctx context.Context, logger *logger.Logger, namespace string, or
...
+	split := strings.Split(orb, "@")
+	if len(split) > 1 {
+		return strings.Join(split, "@")
```
Comment (DanToml) [inline api/api.go]: is this not the same as `return orb`?

File: api/api.go [hunk 4]
```diff
@@ -730,28 +730,42 @@ func OrbPromote(ctx context.Context, logger *logger.Logger, namespace string, or
 	return &response.PromoteOrb.OrbPromoteResponse.Orb, err
 }
 
+func orbVersionRef(orb string) string {
+	split := strings.Split(orb, "@")
+	if len(split) > 1 {
```
Comment (DanToml) [inline api/api.go]: Should it be an error if there is more than 1 `@`?

File: api/api.go (hunk 1 above)
Comment (eric-hu) [inline api/api.go]: Can you add some tests for `orbVersionRef` to demonstrate the use cases?

File: cmd/orb.go [hunk 5]
```diff
@@ -115,14 +115,15 @@ func newOrbCommand() *cobra.Command {
...
 		Short:       "Show the source of an orb",
 		RunE:        showSource,
-		Args:        cobra.ExactArgs(2),
+		Args:        cobra.MaximumNArgs(2),
 		Annotations: make(map[string]string),
 	}
-	sourceCommand.Annotations["NAMESPACE"] = orbAnnotations["NAMESPACE"]
-	sourceCommand.Annotations["ORB"] = orbAnnotations["ORB"]
+	sourceCommand.Annotations["ORB"] = "The name of your orb, including namespace (i.e. circleci/rails)"
+	sourceCommand.Example = `  circleci orb source circleci/python@0.1.4 # grab the source at version 0.1.4
```
Comment (eric-hu) [inline cmd/orb.go]: Nice example 👍

File: api/api.go [hunk 6]
```diff
@@ -730,28 +730,42 @@ func OrbPromote(ctx context.Context, logger *logger.Logger, namespace string, or
...
+	}
+
+	// If no version was supplied, prepend @volatile to the reference
+	return fmt.Sprintf("%s@%s", split[0], "volatile")
+}
+
 // OrbSource gets the source or an orb
-func OrbSource(ctx context.Context, logger *logger.Logger, namespace string, orb string) (string, error) {
-	name := namespace + "/" + orb
+func OrbSource(ctx context.Context, logger *logger.Logger, orb string) (string, error) {
```
Comment (eric-hu) [inline api/api.go]: `orb` would be more accurately named `orbRef`

**Pull request merged**